hardware_interface/src/component_parser.cpp

```diff
@@ -132,20 +132,25 @@ std::unordered_map<std::string, std::string> parse_parameters_from_xml(
 /**
  * \brief Search XML snippet for definition of interfaceTypes.
  *
- * \param interfaces_it pointer to the interator over interfaces
+ * \param interfaces_it pointer to the iterator over interfaces
  * \param interfaceTag interface type tag (command or state)
  * \return std::vector< std::__cxx11::string > list of interface types
  * \throws std::runtime_error if the interfaceType text not set in a tag
  */
-std::vector<std::string> parse_interfaces_from_xml(
+std::vector<components::InterfaceInfo> parse_interfaces_from_xml(
   const tinyxml2::XMLElement * interfaces_it, const char * interfaceTag)
 {
-  std::vector<std::string> interfaces;
+  std::vector<components::InterfaceInfo> interfaces;
 
+  // TODO(andyz): parse optional attributes like min_value/max_value
```

Comment:
@bmagyar do we want to allow this?

I would prefer to be very strict here with `min` and `max` for the name of the paramters.

Reply:
I think we can stick with "min" and "max". I went with "min_value" and "max_value" because that's the example Bence linked me to.  https://github.com/ros-controls/roadmap/issues/23